File: Frontend-BAP/screens/ProcurementDetails.js
import {
  View,
  Text,
  TouchableOpacity,
  Image,
  SafeAreaView,
  ScrollView,
} from 'react-native';
import React from 'react';

import backArrow from '../assets/left-arrow.png';
import tw from 'tailwind-react-native-classnames';
import { procurementdata } from '../constants/procurementdetails';
import { useNavigation } from '@react-navigation/native';
import { images } from '../constants/images';
import { Dimensions } from 'react-native';

const ProcurementDetails = () => {
  const navigation = useNavigation();
  const windowHeight = Dimensions.get('window').height;

  const item = procurementdata;
  console.log(item);
  return (
    <SafeAreaView className='flex-1'>
      <ScrollView>
        <View
          style={tw`flex-row justify-start items-center p-1`}
          className='bg-[#B2FFB7]'
        >
          <TouchableOpacity onPress={() => navigation.navigate('Procurements')}>
            <Image source={backArrow} style={tw`w-6 h-6 p-1`} />
          </TouchableOpacity>
          <Text style={tw`text-xl font-semibold p-2`}>{item.farmerName}</Text>
        </View>
        {/* <Header title={item.farmerName} page="Home"/> */}
        <View style={tw`bg-white p-4 rounded-lg border-gray-300 border-2 m-2`}>
          <View
            style={tw`flex-row justify-between border-b-2 border-gray-400 pb-2 mb-2`}
          >
            <View style={tw`flex-row justify-start`}>
              <Text style={tw`text-sm text-gray-500`}>Date:</Text>
              {/* <Text style={tw`text-lg font-bold`}>{item.date}</Text> */}
              <Text className='text-md'>{item.date}</Text>
            </View>
            <TouchableOpacity
              // style={[tw`flex-row justify-end border-2 rounded-lg p-1`, { borderColor: '#2B9846' }]}
              style={tw`flex-row justify-end border-2 rounded-lg`}
              className='border-[#2B9846] px-2 py-1'
            >
              <Text
                style={tw`text-sm font-semibold`}
                className='text-[#2B9846]'
              >
                Edit
              </Text>
            </TouchableOpacity>
          </View>
          <View className='border-b border-gray-300 w-full mt-2' />

          <Text className='text-md font-bold'>Farmer Details</Text>
          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Farmer Name</Text>
            {/* <Text style={tw`text-lg font-bold`}>{item.farmerName}</Text> */}
            <Text className='text-md font-bold'>{item.farmerName}</Text>
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Farmer No.</Text>
            {/* <Text style={tw`text-lg font-bold`}>{item.farmerMobile}</Text> */}
            <Text className='text-md font-bold'>{item.farmerMobile}</Text>
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Crop Name</Text>
            {/* <Text style={tw`text-md font-bold`}>{item.cropName}</Text> */}
            <Text className='text-md font-bold'>{item.cropName}</Text>
            {/* <Text style={tw`text-md font-bold`}>{item.cropName}</Text> */}
          </View>
          <View className='border-b border-gray-300 w-full mt-2' />

          <Text className='text-md font-bold'>Crop Details</Text>
          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Farmer Name</Text>
            {/* <Text style={tw`text-lg font-bold`}>{item.farmerName}</Text> */}
            <Text className='text-md font-bold'>{item.cropName}</Text>
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Farmer No.</Text>
            {/* <Text style={tw`text-lg font-bold`}>{item.farmerMobile}</Text> */}
            <Text className='text-md font-bold'>{item.farmerMobile}</Text>
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Crop Name</Text>
            {/* <Text style={tw`text-md font-bold`}>{item.cropName}</Text> */}
            <Text className='text-md font-bold'>{item.cropName}</Text>
            {/* <Text style={tw`text-md font-bold`}>{item.cropName}</Text> */}
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Variety</Text>
            <Text style={tw`text-sm font-bold`}>{item.variety}</Text>
          </View>
          <Text style={tw`text-sm text-gray-500 font-semibold`}>
            Crop Photo
          </Text>

          <Image
            source={item.cropPhoto}
            resizeMode='contain'

            style={{
              width: '100%',
              height: windowHeight * 0.25,
              borderRadius: 10,
            }}
          />

          <View className='border-b border-gray-300 w-full mt-2' />
          <Text className='text-md font-bold'>Delivery Person Details</Text>
          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Delivery Person Name</Text>
            {/* <Text style={tw`text-lg font-bold`}>{item.farmerName}</Text> */}
            <Text className='text-md font-bold'>{item.deliveryPersonName}</Text>
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`} className=''>
              Delivery Person Mobile
            </Text>
            <Text className='text-md font-bold'>
              {item.deliveryPersonMobile}
            </Text>
          </View>

          <View style={tw`flex-row justify-between mb-2`}>
            <Text style={tw`text-sm text-gray-500`}>Vehicle No.</Text>
            <Text style={tw`text-sm font-bold`}>{item.tractorNo}</Text>
          </View>

          <Text style={tw`text-sm text-gray-500 font-semibold`}>
            Vehicle Photo
          </Text>

          {/* <Image
            source={images.tractorPhoto}
            resizeMode='contain'
            // style={tw`w-full h-200`}

            // style={{ width: '100%', height: 200, borderRadius: 10 }}
            style={{
              width: '100%',
              height: windowHeight * 0.25,
              borderRadius: 10,
            }}
          /> */}
          <Image
            source={item.vehiclePhoto}
            resizeMode='contain'
           
            style={{
              width: '100%',
              height: windowHeight * 0.25,
              borderRadius: 10,
            }}
          />
        </View>

        <View className='my-1'>
          <TouchableOpacity
            // style={[tw`p-2 rounded-lg my-1`, { backgroundColor: '#2B9846' }]}
            style={tw`p-2 rounded-lg mx-2`}
            className='bg-[#2B9846]'
            // onPress={()=>navigation.navigate('')}
          >
            <Text
              style={tw`text-white text-sm font-semibold justify-center items-center flex`}
            >
              Complete the form
            </Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default ProcurementDetails;

File: demo/lib/screens/procurement_details.dart
import 'package:flutter/material.dart';

class ProcurementDetails extends StatelessWidget {
  const ProcurementDetails({super.key});

  @override
  Widget build(BuildContext context) {
    final double windowHeight = MediaQuery.of(context).size.height;

    // Dummy procurement data (replace with your model or API data)
    final Map<String, dynamic> item = {
      "date": "05/03/2024",
      "farmerName": "Sunita Singh",
      "farmerMobile": "8493021573",
      "cropName": "Wheat",
      "variety": "High Yield",
      "cropPhoto": "assets/images/crop.png",
      "deliveryPersonName": "Ramesh Kumar",
      "deliveryPersonMobile": "9876543210",
      "tractorNo": "TN-12-3456",
      "vehiclePhoto": "assets/images/tractor.png",
    };

    return SafeArea(
      child: Scaffold(
        body: SingleChildScrollView(
          child: Column(
            children: [
              // Header
              Container(
                color: const Color(0xFFB2FFB7),
                padding: const EdgeInsets.all(8),
                child: Row(
                  children: [
                    IconButton(
                      icon: const Icon(Icons.arrow_back, size: 24),
                      onPressed: () {
                        Navigator.pop(context);
                      },
                    ),
                    Text(
                      item['farmerName'],
                      style: const TextStyle(
                        fontSize: 18,
                        fontWeight: FontWeight.w600,
                      ),
                    )
                  ],
                ),
              ),

              // Main card
              Container(
                margin: const EdgeInsets.all(8),
                padding: const EdgeInsets.all(12),
                decoration: BoxDecoration(
                  color: Colors.white,
                  border: Border.all(color: Colors.grey.shade300, width: 2),
                  borderRadius: BorderRadius.circular(10),
                ),
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    // Date & Edit
                    Row(
                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                      children: [
                        Row(
                          children: [
                            const Text(
                              "Date: ",
                              style: TextStyle(
                                fontSize: 14,
                                color: Colors.grey,
                              ),
                            ),
                            Text(item['date'],
                                style: const TextStyle(fontSize: 14)),
                          ],
                        ),
                        OutlinedButton(
                          style: OutlinedButton.styleFrom(
                            side: const BorderSide(color: Color(0xFF2B9846)),
                          ),
                          onPressed: () {},
                          child: const Text(
                            "Edit",
                            style: TextStyle(
                              fontSize: 14,
                              color: Color(0xFF2B9846),
                              fontWeight: FontWeight.w600,
                            ),
                          ),
                        )
                      ],
                    ),

                    const Divider(thickness: 1, color: Colors.grey),

                    // Farmer Details
                    const Text(
                      "Farmer Details",
                      style:
                          TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
                    ),
                    infoRow("Farmer Name", item['farmerName']),
                    infoRow("Farmer No.", item['farmerMobile']),
                    infoRow("Crop Name", item['cropName']),

                    const Divider(thickness: 1, color: Colors.grey),

                    // Crop Details
                    const Text(
                      "Crop Details",
                      style:
                          TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
                    ),
                    infoRow("Crop Name", item['cropName']),
                    infoRow("Variety", item['variety']),
                    const Text(
                      "Crop Photo",
                      style: TextStyle(
                        fontSize: 14,
                        fontWeight: FontWeight.w600,
                        color: Colors.grey,
                      ),
                    ),
                    const SizedBox(height: 6),
                    Image.asset(
                      item['cropPhoto'],
                      height: windowHeight * 0.25,
                      width: double.infinity,
                      fit: BoxFit.contain,
                    ),

                    const Divider(thickness: 1, color: Colors.grey),

                    // Delivery Person
                    const Text(
                      "Delivery Person Details",
                      style:
                          TextStyle(fontSize: 16, fontWeight: FontWeight.bold),
                    ),
                    infoRow("Delivery Person Name", item['deliveryPersonName']),
                    infoRow(
                        "Delivery Person Mobile", item['deliveryPersonMobile']),
                    infoRow("Vehicle No.", item['tractorNo']),

                    const Text(
                      "Vehicle Photo",
                      style: TextStyle(
                        fontSize: 14,
                        fontWeight: FontWeight.w600,
                        color: Colors.grey,
                      ),
                    ),
                    const SizedBox(height: 6),
                    Image.asset(
                      item['vehiclePhoto'],
                      height: windowHeight * 0.25,
                      width: double.infinity,
                      fit: BoxFit.contain,
                    ),
                  ],
                ),
              ),

              // Complete Button
              Container(
                margin: const EdgeInsets.symmetric(horizontal: 8, vertical: 4),
                width: double.infinity,
                child: ElevatedButton(
                  style: ElevatedButton.styleFrom(
                    backgroundColor: const Color(0xFF2B9846),
                    padding: const EdgeInsets.symmetric(vertical: 12),
                    shape: RoundedRectangleBorder(
                      borderRadius: BorderRadius.circular(8),
                    ),
                  ),
                  onPressed: () {},
                  child: const Text(
                    "Complete the form",
                    style: TextStyle(
                      color: Colors.white,
                      fontSize: 14,
                      fontWeight: FontWeight.w600,
                    ),
                  ),
                ),
              )
            ],
          ),
        ),
      ),
    );
  }

  // Reusable Info Row
  Widget infoRow(String label, String value) {
    return Padding(
      padding: const EdgeInsets.symmetric(vertical: 6),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Text(
            label,
            style: const TextStyle(fontSize: 14, color: Colors.grey),
          ),
          Text(
            value,
            style: const TextStyle(fontSize: 14, fontWeight: FontWeight.w600),
          ),
        ],
      ),
    );
  }
}
